### ISSUE TITLE
Glitchy directional stopper

### ISSUE THREAD

**Issue by Kruimeldief** [ddnet/ddnet#5560]
Server: DDNet Pol - Novice (146.59.12.194:8311)
Map: Pados 3
Client: DDNet 16.2
Anti-ping = OFF

Somehow I can't reproduce it in a test map with any directional stopper. But am able to reproduce it in _Pados 3_ (same server), meaning restarting the client and playing the map again.

Ps. video is a bit weirdly cropped.

https://user-images.githubusercontent.com/55532712/177382315-a1cd4499-548c-425f-9279-dc371af3d350.mp4

**Comment by C0D3D3V** [comment]
And what is now the problem?

**Comment by C0D3D3V** [comment]
![grafik](https://user-images.githubusercontent.com/14315968/177384789-5274153d-e9c6-4768-bfa5-c68a0b86100b.png)
just a normal vertical flipped stopper... they act a little different then rotated stoppers. But as log as you can not skip them easily I see no problem (You can skip every stopper with enough speed.).

**Comment by def-** [comment]
Non-fluent bouncing motion sounds like it could be prediction related. @trml

**Comment by C0D3D3V** [comment]
> Edit: replacing that single stopper with a new (same?) stopper facing downwards fixed the non-fluent bouncing. :\

You probably replaced it with a rotated stopper (vertically and horizontally flipped), as I said they work differently, so not the same as a only vertically flipped stopper. I do not see a non-fluent bouncing motion in your video. But it could maybe be possible that the prediction code for the vertically flipped stopper is wrong

**Comment by C0D3D3V** [comment]
> If this is by design, then this issue can be disregarded and closed.

Yes that is by design XD a really strange design to be honest.

**Comment by trml** [comment]
Seems like the prediction is off, since the effect increases with higher ping, but not sure why. Also, from looking at moverestrictions code I can't tell why a vertically flipped stopper would behave differently from a regular downwards facing one in the first place.

**Comment by Marmare314** [comment]
I cannot tell yet why they behave differently either, but I noticed that applying the restriction directly after it is retrieved (as is done in the prediction code) fixes the issue.

**Comment by Marmare314** [comment]
Alright so the difference seems to be that on flipped tiles `Index` is -1 for a few ticks and thus the HandleTiles function is exited before the velocity is clamped. With rotated tiles this does not appear to happen.

**Comment by AssassinTee** [comment]
This is well known and checked for by the ddnet bot on map submissions
